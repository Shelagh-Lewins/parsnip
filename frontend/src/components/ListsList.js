// List of all lists

import React from 'react';
import List from './List';

import { LIST_IS_PUBLIC_VALUES } from '../constants';
import { LIST_IS_PUBLIC_TEXTS } from '../constants';

const ListsList = props => {
	const index = LIST_IS_PUBLIC_VALUES.indexOf(props.is_public === 'true');
	const headerText = LIST_IS_PUBLIC_TEXTS[index];

	return (
		<div className="lists-list">
			<div className="lists-list-title">
				<strong>{headerText}</strong>
			</div>
			{props.lists.map(list => (
				<List
					key={list.id}
					list={list}
					onIsPublicChange={props.onIsPublicChange}
					onDeleteList={props.onDeleteList}
				/>
			))}
		</div>
	);
};

export default ListsList;
