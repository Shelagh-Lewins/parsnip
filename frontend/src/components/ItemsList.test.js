import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ListsList from './ListsList';
import toJson from 'enzyme-to-json';

Enzyme.configure({ 'adapter': new Adapter() });

describe('the ListsList component', () => {
	it('should render an is_public', () => {
		const wrapper = shallow(<ListsList lists={[]} />);

		expect(wrapper.find('strong').text()).toEqual('Private');
	});

	it('should render a List component for each list', () => {
		const lists = [
			{ 'id': 'abc', 'title': 'a list', 'description': 'some stuff', 'is_public': true },
			{ 'id': 'def', 'title': 'another list', 'description': 'some more stuff', 'is_public': true },
		];

		const wrapper = shallow(<ListsList is_public="false" lists={lists} />);

		expect(wrapper.find('List').length).toEqual(2);
	});
});

describe('it should match the last snapshot without lists', () => {
	const wrapper = shallow(<ListsList is_public="false" lists={[]} />);
	expect(toJson(wrapper)).toMatchSnapshot();

	it('should match the last snapshot with lists', () => {
		const lists = [
			{ 'id': 'abc', 'title': 'a list', 'description': 'some stuff' },
			{ 'id': 'def', 'title': 'another list', 'description': 'some more stuff' },
		];

		const wrapper = shallow(<ListsList is_public="false" lists={lists} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
