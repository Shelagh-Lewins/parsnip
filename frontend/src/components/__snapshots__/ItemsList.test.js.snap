// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="lists-list"
>
  <div
    className="lists-list-title"
  >
    <strong>
      Private
    </strong>
  </div>
</div>
`;

exports[`it should match the last snapshot without lists should match the last snapshot with lists 1`] = `
<div
  className="lists-list"
>
  <div
    className="lists-list-title"
  >
    <strong>
      Private
    </strong>
  </div>
  <List
    key="abc"
    list={
      Object {
        "description": "some stuff",
        "id": "abc",
        "title": "a list",
      }
    }
  />
  <List
    key="def"
    list={
      Object {
        "description": "some more stuff",
        "id": "def",
        "title": "another list",
      }
    }
  />
</div>
`;
